lib/PrintAsObjC/PrintAsObjC.cpp

```diff
@@ -2634,13 +2617,70 @@ class ModuleWriter {
     return import == importer->getImportedHeaderModule();
   }
 
+  static void getClangSubmoduleReversePath(SmallVectorImpl<StringRef> &path,
+                                           const clang::Module *module) {
+    // FIXME: This should be an API on clang::Module.
+    do {
+      path.push_back(module->Name);
+      module = module->Parent;
+    } while (module);
+  }
+
+  static int compareImportModulesByName(const ImportModuleTy *left,
+                                        const ImportModuleTy *right) {
+    auto *leftSwiftModule = left->dyn_cast<ModuleDecl *>();
+    auto *rightSwiftModule = right->dyn_cast<ModuleDecl *>();
+
+    if (leftSwiftModule && !rightSwiftModule)
+      return -compareImportModulesByName(right, left);
+
+    if (leftSwiftModule && rightSwiftModule)
+      return leftSwiftModule->getName().compare(rightSwiftModule->getName());
+
+    auto *leftClangModule = left->get<const clang::Module *>();
+    assert(leftClangModule->isSubModule());
```

Comment:
Dumb question: why is it always a submodule?

Reply:
Ah, because PrintAsObjC uses `swift::ModuleDecl` for all top-level modules, even imported ones. I included an assertion comment down on line 2695 but I should probably include it here as well.